apps/chat/bots.py

```diff
@@ -278,22 +279,17 @@ def filter_ai_messages(self) -> bool:
 
 
 class PipelineBot:
-    def __init__(self, session: ExperimentSession):
-        self.experiment = session.experiment_version
+    def __init__(self, session: ExperimentSession, experiment: Experiment):
+        self.experiment = experiment
         self.session = session
+        self.ai_message_id = None
 
     def process_input(self, user_input: str, save_input_to_history=True, attachments: list["Attachment"] | None = None):
         output = self.experiment.pipeline.invoke(
             PipelineState(messages=[user_input], experiment_session=self.session), self.session
         )
+        self.ai_message_id = output["ai_message_id"]
         return output["messages"][-1]
 
-    def get_ai_message_id(self) -> int | None:
-        last_ai_message = (
-            ChatMessage.objects.filter(chat=self.session.chat, message_type=ChatMessageType.AI.value)
-            .values("id")
-            .last()
-        )
-        if not last_ai_message:
-            return None
-        return last_ai_message["id"]
+    def get_ai_message_id(self):
```

Comment:
We need a better way of getting the AI message. I'm storing it on the state and reading it from there, but is that the best way to do it?